File: src/main/java/dev/klomptech/jbark/parser/keywords/ControlFlowKeywordGroups.java
package dev.klomptech.jbark.parser.keywords;

import java.util.List;

public final class ControlFlowKeywordGroups {

  public final List<String> ifStartWords;
  public final List<String> whileWords;
  public final List<String> untilWords;
  public final List<String> forWords;
  public final List<String> eachWords;
  public final List<String> fromWords;
  public final List<String> thenWords;
  public final List<String> otherwiseWords;
  public final List<String> elseWords;
  public final List<String> buryWords;
  public final List<String> breakWords;
  public final List<String> continueWords;
  public final List<String> expectsWords;
  public final List<String> trickReturnsWords;

  public ControlFlowKeywordGroups(final KeywordRegistry registry) {
    ifStartWords = registry.phraseWords("if", "when", "whenever", "should");
    whileWords = registry.phraseWords("while", "during");
    untilWords = registry.phraseWords("until", "till");
    forWords = registry.phraseWords("for");
    eachWords = registry.phraseWords("each", "every");
    fromWords = registry.phraseWords("from");
    thenWords = registry.phraseWords("then", "do");
    otherwiseWords = registry.phraseWords("otherwise", "instead");
    elseWords = registry.phraseWords("else");
    buryWords =
        registry.phraseWords(
            "bury",
            "end",
            "goodnight",
            "enough",
            "bedtime",
            "lightsout",
            "hush",
            "done",
            "finished");
    breakWords = registry.phraseWords("heel", "stop", "stay", "halt");
    continueWords = registry.phraseWords("again", "repeat", "resume", "onward");
    expectsWords = registry.phraseWords("expects", "expect");
    trickReturnsWords = registry.phraseWords("returns", "return");
  }
}

File: pbark/pbark/parser/keywords/control_flow_keyword_groups.py
from __future__ import annotations

from pbark.parser.keywords.keyword_registry import KeywordRegistry


class ControlFlowKeywordGroups:
    def __init__(self, registry: KeywordRegistry) -> None:
        self.if_start_words = registry.phrase_words("if", "when", "whenever", "should")
        self.while_words = registry.phrase_words("while", "during")
        self.until_words = registry.phrase_words("until", "till")
        self.for_words = registry.phrase_words("for")
        self.each_words = registry.phrase_words("each", "every")
        self.from_words = registry.phrase_words("from")
        self.then_words = registry.phrase_words("then", "do")
        self.otherwise_words = registry.phrase_words("otherwise", "instead")
        self.else_words = registry.phrase_words("else")
        self.bury_words = registry.phrase_words(
            "bury", "end", "goodnight", "enough", "bedtime", "lightsout", "hush", "done", "finished"
        )
        self.break_words = registry.phrase_words("heel", "stop", "stay", "halt")
        self.continue_words = registry.phrase_words("again", "repeat", "resume", "onward")
        self.expects_words = registry.phrase_words("expects", "expect")
        self.trick_returns_words = registry.phrase_words("returns", "return")
